backend/de.metas.business/src/main/java/de/metas/product/impl/ProductDAO.java

```diff
@@ -640,12 +640,15 @@ public Optional<ProductId> getProductIdByGTIN(@NonNull final GTIN gtin, @NonNull
 	}
 
 	@Override
-	public Optional<ProductId> getProductIdByValueStartsWith(@NonNull final String valuePrefix, @NonNull final ClientId clientId)
+	public Optional<ProductId> getProductIdByEAN13CodeOrValue(@NonNull final String ean13code, @NonNull final ClientId clientId)
 	{
 		final ImmutableSet<ProductId> productIds = queryBL.createQueryBuilderOutOfTrx(I_M_Product.class)
 				.addOnlyActiveRecordsFilter()
 				.addEqualsFilter(I_M_Product.COLUMNNAME_AD_Client_ID, clientId)
-				.addStringStartsWith(I_M_Product.COLUMNNAME_Value, valuePrefix)
+				.filter(queryBL.createCompositeQueryFilter(I_M_Product.class)
+								.setJoinOr()
+								.addStringLikeFilter(I_M_Product.COLUMNNAME_UPC, "__"+ ean13code.substring(0, 4) + "%", true)
```

Comment:
what exactly is this `ean13code` string?

Reply:
why substring 0, 4?
(above somewhere it was 2,6) ....

wondering why not having some EAN13 small class to abstract these things

Reply:
> why substring 0, 4?
They are the first 4 characters from de.metas.handlingunits.qrcodes.ean13.EAN13HUQRCode#productNo